src/modules/python/pbs_hooks/PBS_cray_jacs.PY

```diff
@@ -0,0 +1,339 @@
+        cls.config = config
+
+    @classmethod
+    def get_event(cls):
+        """
+        Fetch the event if it hasn't already been fetched.
+        Return the event
+        """
+        if not cls.event:
+            cls.event = pbs.event()
+        return cls.event
+
+    @classmethod
+    def get_config(cls):
+        """
+        Load the config if it hasn't already been loaded.
+        Return the config
+        """
+        if not cls.config:
+            cls.load_config()
+        return cls.config
+
+    @staticmethod
+    def build_path(resource, jobid=None):
+        """
+        Given a jobid and a resource type build the path.
+        If jobid is none, build a path to the collection
+        requests_unixsockets requires the socket path to be percent-encoded
+        """
+        log_function_name()
+        cfg = HookHelper.get_config()
+
+        if resource not in cfg['resources']:
+            log_with_caller(pbs.EVENT_ERROR, 'Invalid resource type %s' %
+                            resource)
+            raise RejectError()
+
+        path = 'http+unix://%s%s%s%s' % (
+            urllib.quote(cfg['unix_socket_file'], safe=''),
+            cfg['version_uri'],
+            cfg['resources'][resource],
+            ('/' + jobid) if jobid else ''
+        )
+        log_with_caller(pbs.EVENT_DEBUG3, 'path is %s' % path)
+        return path
+
+
+def post(url, json=None, **kwargs):
+    """
+    Wrapper to requests.post
+
+    Logs before and after
+    """
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Sending POST to %s' % url)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Sending JSON: %s' % JSON.dumps(json))
+    r = requests.post(url, json=json, **kwargs)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received status code = %s' % r.status_code)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received text %s' % r.text)
+    return r
+
+
+def get(url, params=None, **kwargs):
+    """
+    Wrapper to requests.get
+
+    Logs before and after
+    """
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Sending GET to %s' % url)
+    if params:
+        pbs.logmsg(pbs.EVENT_DEBUG2, 'Sending params: %s' % params)
+    r = requests.get(url, params=params, **kwargs)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received status code = %s' % r.status_code)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received text %s' % r.text)
+    return r
+
+
+def delete(url, **kwargs):
+    """
+    Wrapper to requests.delete
+
+    Logs before and after
+    """
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Sending DELETE to %s' % url)
+    r = requests.delete(url, **kwargs)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received status code = %s' % r.status_code)
+    pbs.logmsg(pbs.EVENT_DEBUG2, 'Received text %s' % r.text)
+    return r
+
+
+def caller_name(frames=1):
+    """
+    Return the name of the nth calling function or method.
+    """
+    return str(sys._getframe(frames).f_code.co_name)
+
+
+def log_function_name():
+    """
+    Log the caller's name
+    """
+    pbs.logmsg(pbs.EVENT_DEBUG4, '%s: Method called' % caller_name(2))
```

Comment:
Since you aren't the only pbs hook that uses this terminology in the logs, you might want to put the hook name in the log message.